go/cmd/dolt/commands/branch.go

```diff
@@ -243,159 +324,186 @@ func printAllDatasets(ctx context.Context, dEnv *env.DoltEnv) int {
 	return 0
 }
 
-func moveBranch(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults, usage cli.UsagePrinter) int {
-	if apr.NArg() != 2 {
-		usage()
-		return 1
+// generateBranchSql returns the query that will call the `DOLT_BRANCH` stored procedure.
+// This function assumes that the inputs are validated branch names or flags, neither of which can contain quotes.
```

Comment:
Tracking back up the stack, it's not immediately obvious that all validation for these args is done at line 126. Rather than say this "function assumes" I'd say it required prior validation - and callout where that is happening above. I'm not wild about this though; it's frail and will get broken by someone in the future.

Ultimately the interpolate query solution that Steph is working on is a better solution all around. Can you take a look at that - should be safer and more future proof.

Reply:
Done.